src/reconstruction/comparison/modelBorgifier/src/compareCbModels.m

```diff
@@ -0,0 +1,551 @@
+% This file is published under Creative Commons BY-NC-SA.
+%
+% Please cite:
+% Sauls, J. T., & Buescher, J. M. (2014). Assimilating genome-scale 
+% metabolic reconstructions with modelBorgifier. Bioinformatics 
+% (Oxford, England), 30(7), 1036?8. http://doi.org/10.1093/bioinformatics/btt747
+%
+% Correspondance:
+% johntsauls@gmail.com
+%
+% Developed at:
+% BRAIN Aktiengesellschaft
+% Microbial Production Technologies Unit
+% Quantitative Biology and Sequencing Platform
+% Darmstaeter Str. 34-36
+% 64673 Zwingenberg, Germany
+% www.brain-biotech.de
+%
+function [Cmodel, Tmodel, score, Stats] = compareCbModels(Cmodel, Tmodel)
+% compareCbModels Calculate similarity of two Cb models. These models must
+% be in a COBRA-like format.
+%
+% USAGE:
+%    [Cmodel, Tmodel, score, Stats] = compareCbModels(Cmodel, Tmodel)
+%
+% INPUTS:
+%    Cmodel:    Model to be compared to the template. In COBRA format.
+%
+% OPTIONAL INPUTS:
+%    Tmodel:    Template model. In COBRA format.
+%
+% OUTPUTS:
+%    Cmodel:    Appended Cb comparison model with additional information to
+%               faciliate comparison.
+%    Tmodel:    Appended Cb template model with additional information to
+%               facilitate comparision.
+%    score      Every individual score, split by number of comparison factors.
+%               Size = MxNxF
+%    Stats      Structure of statsistics. Including
+%    scoreTotal    MxN matrix of normalized (0 to 1) scores with M equal 
+%                  to the number of reactions in Cmodel and N equal to the 
+%                  number of reactions in Tmodel.
+%
+% CALLS:
+%    addComparisonInfo:     Subfunction. 
+%    compareMetInfo:        Subfunction.
+%    optimalScores
+%    mapMatch
+%    buildRxnEquations
+%    fixNames
+%
+% CALLED BY:
+%    driveModelBorgifier
+%
+%NOTES:
+%    Use readCbModel to create a COBRA model from a SBML formatted .xml or an
+%    appropriate scipt to create the model from an .xls or other format.
+%    Run script through verifyModel to make sure that is has all the correct 
+%    information for compareCbModels.
+%
+
+%% Declare oft used variables and set up scoring system and arrays.
+nCrxns = length(Cmodel.rxns);
+compCrxns = 1:nCrxns ; % Comparison C reactions
+nTrxns = length(Tmodel.rxns) ; % Number of T reactions
+
+% Names of all models currently in Tmodel.
+modelNames = fieldnames(Tmodel.Models) ;
+
```

Comment:
As above, entries can have multiple ids assigned - separated by ; -, particularily prevalent with ChEBI IDs, so it might be necessary to check all ids after splitting them.